x/ibc/23-commitment/types/merkle.go

```diff
@@ -366,7 +366,7 @@ func convertProofs(mproof MerkleProof) ([]*ics23.CommitmentProof, error) {
 
 // Empty returns true if the root is empty
 func (proof MerkleProof) Empty() bool {
-	return proof.Proof.Equal(nil) || proof.Equal(MerkleProof{}) || proof.Proof.Equal(nil) || proof.Proof.Equal(merkle.Proof{})
+	return proto.Equal(proof.Proof, nil) || proto.Equal(&proof, &MerkleProof{}) || proto.Equal(proof.Proof, nil) || proto.Equal(proof.Proof, &tmmerkle.ProofOps{})
```

Comment:
this is outside the scope of this pr, but there is a duplicate check here with `proto.Equal(proof.Proof, nil)`

Reply:
I think this is a bug, I think the first check should be `proto.Equal(proof, nil)`